### PR TITLE
[test] connect UARTs/I2Cs to pins and TX data on I2Cs in power virus test

### PR DESCRIPTION
**Pull request by timothytrippel**
This PR contains several commits that:
1. connect UARTs/I2Cs to MIOs
2. load I2C FIFOs with data and enable them to transmit data
3. cleanup the I2C testutils

### CODE REVIEW

File: sw/device/tests/power_virus_systemtest.c [hunk 1]
```diff
@@ -230,15 +245,87 @@ static void init_peripheral_handles(void) {
...
+                                       kTopEarlgreyPinmuxPeripheralInUart3Rx,
+                                       kTopEarlgreyPinmuxInselIoa0));
+  CHECK_DIF_OK(dif_pinmux_output_select(&pinmux, kTopEarlgreyPinmuxMioOutIoa0,
+                                        kTopEarlgreyPinmuxOutselConstantHighZ));
+  CHECK_DIF_OK(dif_pinmux_output_select(&pinmux, kTopEarlgreyPinmuxMioOutIoa1,
+                                        kTopEarlgreyPinmuxOutselUart3Tx));
+
+  // I2C0:
+  //    SCL on IOA7
+  //    SDA on IOA8
```
Comment (a-will) [inline sw/device/tests/power_virus_systemtest.c:290]: If you're wanting hyperdebug compatibility, these pins are swapped:

```suggestion
  //    SDA on IOA7
  //    SCL on IOA8
```

File: sw/device/tests/power_virus_systemtest.c [hunk 2]
```diff
@@ -230,15 +245,87 @@ static void init_peripheral_handles(void) {
...
+                                       kTopEarlgreyPinmuxPeripheralInI2c0Sda,
+                                       kTopEarlgreyPinmuxInselIoa8));
+  CHECK_DIF_OK(dif_pinmux_output_select(&pinmux, kTopEarlgreyPinmuxMioOutIoa7,
+                                        kTopEarlgreyPinmuxOutselI2c0Scl));
+  CHECK_DIF_OK(dif_pinmux_output_select(&pinmux, kTopEarlgreyPinmuxMioOutIoa8,
+                                        kTopEarlgreyPinmuxOutselI2c0Sda));
+
+  // I2C1:
+  //    SCL on IOB9
+  //    SDA on IOB10
```
Comment (a-will) [inline sw/device/tests/power_virus_systemtest.c:304]: For hyperdebug compatibility, swap them:

```suggestion
  //    SDA on IOB9
  //    SCL on IOB10
```

**Review by a-will — APPROVED** [review]

File: sw/device/tests/power_virus_systemtest.c (hunk 1 above)
Comment (timothytrippel) [reply sw/device/tests/power_virus_systemtest.c]: ah, thanks. fixed.

File: sw/device/tests/power_virus_systemtest.c (hunk 2 above)
Comment (timothytrippel) [reply sw/device/tests/power_virus_systemtest.c]: fixed.

**Pull request merged**